site2/docs/functions-develop.md

```diff
@@ -702,3 +716,7 @@ To access metrics created by Pulsar Functions, refer to [Monitoring](deploy-moni
 Pulsar Functions use [Apache BookKeeper](https://bookkeeper.apache.org) as a state storage interface. Pulsar installation, including the local standalone installation, includes deployment of BookKeeper bookies.
 
 Since Pulsar 2.1.0 release, Pulsar integrates with Apache BookKeeper [table service](https://docs.google.com/document/d/155xAwWv5IdOitHh1NVMEwCMGgB28M3FyMiQSxEpjE-Y/edit#heading=h.56rbh52koe3f) to store the `State` for functions. For example, a `WordCount` function can store its `counters` state into BookKeeper table service via Pulsar Functions State API.
+
+States are key-value pairs, where the key is a string and the value is arbitrary binary data - counters are stored as 64-bit big-endian binary values.  Keys are scoped to an individual pulsar function, but shared between all instances of that function.
```

Comment:
```suggestion
States are key-value pairs, where the key is a string and the value is arbitrary binary data - counters are stored as 64-bit big-endian binary values. Thought keys are scoped to an individual Pulsar Function, the keys are shared among all instances of that function.
```

Reply:
s/Thought/Though/ ?

I think the conjunction isn't useful here, as it implies the user already knows the scoping of keys.  I don't think it's mentioned earlier, and without this knowledge I might have guessed they were scoped differently (to the pulsar "namespace" that the function executes within, for example)

So as a user, I just want a statement which answers the question: "What's the scope of the key?"

I would be happy with "Keys are scoped to the pulsar function".  I thought it worth clarifying functions versus function instances, but maybe that's unnecessary.  State storage which *wasn't* shared between function instances wouldn't be very useful.

Reply:
"Though", sorry for the typo.
If you think the conjunction is useful, you can remove it.